hack/packages/package-tools/cmd/vendir-sync.go

```diff
@@ -67,3 +79,33 @@ func syncPackage(packagePath, toolsBinDir string) error {
 	}
 	return nil
 }
+
+// selectPackages will return a map of package names as keys in a given repo. If
+// no repo is provided, all packages will be in the map.
+func selectPackages(pkgVals PackageValues, repoName string) (map[string]struct{}, error) {
```

Comment:
Use filterPackageRepo instead

Reply:
@yharish991, I made use of `filterPackageRepos` and put the changes in a separate commit so you can see what had to change. I did indeed have to make another helper function to look through a slice and look for a particular package. Let me know if you want me to keep the change or revert.

Reply:
I'll resolve this comment. I take from @yharish991 's later [comment](https://github.com/vmware-tanzu/tanzu-framework/pull/1906#pullrequestreview-939707125) that this latest version where `filterPackageRepos` is used is preferable and I don't need to revert to using `selectPackages`.